processing/src/main/java/io/druid/query/QueryMetrics.java

```diff
@@ -107,29 +107,31 @@
  *
  * Making subinterfaces of QueryMetrics for emitting custom dimensions and/or metrics for specific query types
```

Comment:
Instead of referencing TopN, GroupBy and Timeseries as something that should *not* be taken as examples of following this procedure, in the end of the description of this procedure, please reference Search and Select as examples of this procedure implemented. Also please add notes about empty or exception throwing of `query()` and other "pre-query-execution-time" methods to this procedure.